libreyolo/label/dataset.py

```
@@ -0,0 +1,197 @@
+
+def _atomic_write_text(path: Path, text: str) -> None:
+    """Write via a temp file + ``os.replace`` so a label is never half-written."""
+    tmp = path.with_suffix(path.suffix + ".tmp")
+    tmp.write_text(text, encoding="utf-8")
+    os.replace(tmp, path)
+
+
+class DatasetSession:
+    """An open dataset: ordered images across train/val/test + label R/W."""
+
+    def __init__(self, data: str):
```

Ruff (0.15.15)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)